sympy/core/numbers.py

```diff
@@ -829,7 +829,7 @@ def as_coeff_Mul(self, rational=False):
         """Efficiently extract the coefficient of a product."""
         if rational and not self.is_Rational:
             return S.One, self
-        return (self, S.One) if self else (S.One, self)
+        return self, S.One
```

Comment:
It is not very evident why it affects float though. There are only two classes that subclass `Number`, and the one in the `Rational` got overridden, so it could better to override this in `Float`

Reply:
Why is the condition necessary if it's going to be overridden in `Rational` and `Float`?

Reply:
It should reduce to 
```
        if rational:
            return S.One, self
        return self, S.One
```
if it is specialized in `Float`.

Reply:
Oh I see since `Rational` has it's own method we can assume `self` is not rational.  I updated this to

```
        if not rational:
            return self, S.One
        return S.One, self
```
to mirror`as_coeff_Add`.